docs/explanation/remote-build.rst

```diff
@@ -0,0 +1,172 @@
+Remote build
+============
+
+Remote build is a feature in snapcraft that offloads the build process to
+`Launchpad`_'s `build farm`_ and enables developers to build snaps for
+different architectures.
+
+Architectures supported by Launchpad can be found
+:ref:`here<supported-architectures>`.
+
+Open vs closed source
+---------------------
+
+By default, prospective snaps must be open source because the build will be
+publicly available.
+
+Developers are reminded of this by confirming that their project will be
+publicly available when starting a remote build. This prompt can be
+automatically agreed to by passing ``--launchpad-accept-public-upload``.
+
+Closed-source projects can be built using the remote builder. This requires
+the user to create a private Launchpad project and pass the project with the
+``--project <project-name>`` command line argument.
+
+Git repository
+--------------
+
+Projects must be in the top level of a git repository because snapcraft uses
+a git-based workflow to upload projects to Launchpad.
+
+Shallowly cloned repositories are not supported (i.e. ``git clone --depth 1``)
+because git does not support pushing shallow clones.
+
+Versions
+--------
+
+Two versions of the remote-builder are available, the current and the legacy
+remote-builder.
+
+Current
+^^^^^^^
+
+The current remote builder is available for ``core20``, ``core22``, ``core24``,
+and newer snaps.
+
+It does not modify the project or project metadata.
+
+Legacy
+^^^^^^
+
+The "fallback" or legacy version of the remote builder can be used for
+``core20`` and ``core22`` snaps.  It is not available for ``core24`` and newer
+snaps.
+
+The legacy remote builder was deprecated because of its design. It retrieves
+and tarballs remote sources and modifies the project's ``snapcraft.yaml``
+file to point to the local tarballs.
+
+Choosing a remote-builder
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The environment variable ``SNAPCRAFT_REMOTE_BUILD_STRATEGY`` determines which
+remote-builder is used:
+
+* ``disable-fallback`` will use the current remote builder
+* ``force-fallback`` will use the legacy remote builder
+
+If the environment variable is unset, the remote builder will be determined
+by the base:
+
+* ``core22``, ``core24``, and newer snaps will use the current remote builder
+* ``core20`` snaps will use the legacy remote builder by default.
+
+Platforms and architectures
+---------------------------
+
+Remote builds can be orchestrated for multiple platforms and architectures.
+
+Current
+^^^^^^^
+
+``--build-for`` and ``--build-on``
+**********************************
+
+If  ``--build-for`` or ``--build-on`` are provided, Snapcraft will:
+
+# parse the project metadata for ``platforms`` or ``architectures`` keywords
+# create a build plan
+# filter the build plan with ``--build-for`` and ``--platform``
+
+``--build-for`` and ``--platform`` are mutually exclusive keywords.
```

Comment:
I'm confused: the section heading mentions build-for and build-on, where does `--platform` enter here?

Reply:
Whoops, I used the wrong terms in the header.